p2p/transport/webrtc/sdp.go

```diff
@@ -0,0 +1,121 @@
+package libp2pwebrtc
+
+import (
+	"crypto"
+	"fmt"
+	"net"
+
+	"github.com/multiformats/go-multihash"
+)
+
+const clientSDP string = `
+v=0
+o=- 0 0 IN %s %s
+s=-
+c=IN %s %s
+t=0 0
+m=application %d UDP/DTLS/SCTP webrtc-datachannel
+a=mid:0
+a=ice-options:trickle
+a=ice-ufrag:%s
+a=ice-pwd:%s
+a=fingerprint:sha-256 ba:78:16:bf:8f:01:cf:ea:41:41:40:de:5d:ae:22:23:b0:03:61:a3:96:17:7a:9c:b4:10:ff:61:f2:00:15:ad
+a=setup:actpass
+a=sctp-port:5000
+a=max-message-size:16384
+`
+
+func renderClientSdp(addr *net.UDPAddr, ufrag string) string {
+	ipVersion := "IP4"
+	if addr.IP.To4() == nil {
+		ipVersion = "IP6"
+	}
+	return fmt.Sprintf(
+		clientSDP,
+		ipVersion,
+		addr.IP,
+		ipVersion,
+		addr.IP,
+		addr.Port,
+		ufrag,
+		ufrag,
+	)
+}
+
+const serverSDP string = `
+v=0
+o=- 0 0 IN %s %s
+s=-
+t=0 0
+a=ice-lite
+m=application %d UDP/DTLS/SCTP webrtc-datachannel
+c=IN %s %s
+a=mid:0
+a=ice-options:ice2
+a=ice-ufrag:%s
+a=ice-pwd:%s
+a=fingerprint:%s
+a=setup:passive
+a=sctp-port:5000
+a=max-message-size:16384
+a=candidate:1 1 UDP 1 %s %d typ host
+`
+
+func renderServerSdp(addr *net.UDPAddr, ufrag string, fingerprint *multihash.DecodedMultihash) string {
+	ipVersion := "IP4"
+	if addr.IP.To4() == nil {
+		ipVersion = "IP6"
+	}
+	fp := fingerprintToSDP(fingerprint)
+	return fmt.Sprintf(
+		serverSDP,
+		ipVersion,
+		addr.IP,
+		addr.Port,
+		ipVersion,
+		addr.IP,
+		ufrag,
+		ufrag,
+		fp,
+		addr.IP,
+		addr.Port,
+	)
+}
+
+func getSupportedSDPHash(code uint64) (crypto.Hash, bool) {
+	switch code {
+	case multihash.MD5:
```

Comment:
might want to consider having a static `map`ping here?

Reply:
No preference here. I think a switch is fine. It's probably slightly faster, since you don't need to hash the map key.

Reply:
It was previously a map, but considering that these are very few in number and do not change, this would be an O(1) operation with a small overhead.

Reply:
same comment as https://github.com/libp2p/go-libp2p/pull/1655#discussion_r1066754411 applies here.

Reply:
The switch is definitely faster but it is really only nanoseconds of difference.

Reply:
Looks fine now, please feel free to resolve (I do not seem to have the powers to resolve it myself).